### DISCUSSION TITLE
For Optional arguments: Use `?` instead of keyword `optional`

### DISCUSSION THREAD

**Discussion by gargajit** [fastn-stack/fastn#631] [Ideas & RFCs]
Use the `?` symbol at the end of optional arguments, rather than the word `optional`, as it can make the code more concise and easier to read. For example, instead of 

```ftd
-- component foo:
optional caption title:

-- end: foo
```

Use

```ftd
-- component foo:
caption title?:

-- end: foo
```

Suggested by: @saurabh-lohiya

**Comment by amitu** [comment]
@Arpita-Jaiswal this is interesting. Not sure if the `?` belongs to the variable name or to type. Variable name has advantage that is reminds that is it optional at all call sites, so we know when to check.

On the flip-side, maybe we should do generics and do Rust like syntax for Optional, as this syntax does not scale to more complex scenarios.

**Comment by saurabh-lohiya** [comment]
`?` shouldn't belong to the variable name but rather a shorthand to the `optional` keyword
